File: storage-sdk/src/main/java/com/tencent/wink/storage/winkdb/converter/IListTypeConverter.java
package com.tencent.wink.storage.winkdb.converter;

import androidx.annotation.Nullable;
import androidx.room.TypeConverter;



import java.util.List;

/**
 * Created by walkerzpli on 2021/9/22.
 */
public interface IListTypeConverter<T> {

    @TypeConverter
    @Nullable
    String converter(@Nullable List<T> list);

    @TypeConverter
    @Nullable
    List<T> revert(@Nullable String str);

}

File: storage-ktx/src/main/java/com/demo/storage_ktx/converter/IListTypeConverter.kt
package com.demo.storage_ktx.converter

import androidx.room.TypeConverter

/**
 * Created by walkerzpli on 2021/9/22.
 */
interface IListTypeConverter<T> {
    @TypeConverter
    fun converter(list: List<T>?): String?

    @TypeConverter
    fun revert(str: String?): List<T>
}